File: app/src/main/java/com/example/workouttrainingconstruction/ExerciseRecyclerAdapter.java
package com.example.workouttrainingconstruction;

        import android.content.Context;
        import android.view.LayoutInflater;
        import android.view.View;
        import android.view.ViewGroup;
        import android.widget.TextView;
        import androidx.recyclerview.widget.RecyclerView;
        import java.util.List;


public class ExerciseRecyclerAdapter extends RecyclerView.Adapter<ExerciseRecyclerAdapter.ViewHolder> {

        private List<Exercise> mData;
        private LayoutInflater mInflater;
        private ItemClickListener mClickListener;

        ExerciseRecyclerAdapter(Context context, List<Exercise> data) {
            this.mInflater = LayoutInflater.from(context);
            this.mData = data;
        }

        @Override
        public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            View view = mInflater.inflate(R.layout.exercise_item, parent, false);
            return new ViewHolder(view);
        }

        @Override
        public void onBindViewHolder(ViewHolder holder, int position) {
            Exercise exercise = mData.get(position);
            holder.myTextView.setText(exercise.getName());
        }

        @Override
        public int getItemCount() {
            return mData.size();
        }

        public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
            TextView myTextView;

            ViewHolder(View itemView) {
                super(itemView);
                myTextView = itemView.findViewById(R.id.exerciseName);
                itemView.setOnClickListener(this);
            }

            @Override
            public void onClick(View view) {
                if (mClickListener != null) mClickListener.onItemClick(view, getAdapterPosition());
            }
        }

        Exercise getItem(int id) {
            return mData.get(id);
        }

        void setClickListener(ItemClickListener itemClickListener) {
            this.mClickListener = itemClickListener;
        }

        public interface ItemClickListener {
            void onItemClick(View view, int position);
        }
    }

File: app/src/main/java/com/example/workouttrainingconstruction/ExerciseRecyclerAdapter.kt
package com.example.workouttrainingconstruction

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import com.example.workouttrainingconstruction.DataBase.Exercise

class ExerciseRecyclerAdapter internal constructor(context: Context?, data: List<Exercise>) :
    RecyclerView.Adapter<ExerciseRecyclerAdapter.ViewHolder>() {
    private val mData: List<Exercise>
    private val mInflater: LayoutInflater
    private var mClickListener: ItemClickListener? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = mInflater.inflate(R.layout.exercise_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val exercise = mData[position]
        holder.myTextView.text = exercise.name
    }

    override fun getItemCount(): Int {
        return mData.size
    }

    inner class ViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView),
        View.OnClickListener {
        var myTextView: TextView
        override fun onClick(view: View) {
            if (mClickListener != null) mClickListener!!.onItemClick(view, adapterPosition)
        }

        init {
            myTextView = itemView.findViewById(R.id.exerciseName)
            itemView.setOnClickListener(this)
            itemView.setOnLongClickListener {
                val fragmentManager = (itemView.context as AppCompatActivity).supportFragmentManager
                val myFragment: DeleteFragment = DeleteFragment.Companion.newInstance(
                    mData[position].name,
                    "deleteExercise", position, mData[position].id
                )
                Log.d("myTag", Integer.toString(position))
                fragmentManager.beginTransaction().add(R.id.trContainer, myFragment, "deleteFrag")
                    .addToBackStack("deleteFrag")
                    .commit()
                true
            }
        }
    }

    fun getItem(id: Int): Exercise {
        return mData[id]
    }

    fun setClickListener(itemClickListener: ItemClickListener?) {
        mClickListener = itemClickListener
    }

    interface ItemClickListener {
        fun onItemClick(view: View?, position: Int)
    }

    init {
        mInflater = LayoutInflater.from(context)
        mData = data
    }
}